.github/workflows/docker-image.yml

```
@@ -0,0 +1,41 @@
+name: Docker Image CI
+
+on:
+  push:
+    branches: [ "main" ]
+
+jobs:
+
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+
+    # https://github.com/docker/setup-buildx-action
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v3
+
+    # https://github.com/docker/login-action
+    - name: Login to Docker Hub
+      uses: docker/login-action@v3
+      with:
+        username: ${{ vars.DOCKERHUB_USERNAME }}
```

One extra suggestion of changing from vars to secrets here, reasons are two-fold: 1) hide the username being used (I mean: avoid them to be reused across builds); 2) have just one place to configure them.

```suggestion
        username: ${{ secrets.DOCKERHUB_USERNAME }}
```